pkg/cmd/base/options.go

```diff
@@ -58,18 +67,81 @@ func WrapRunE(o CommandOptions, f util.Factory) func(cmd *cobra.Command, args []
 	}
 }
 
+type CobraCompletionFunc func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective)
+type CobraCompletionFuncWithError func(ctx context.Context, manager target.Manager) ([]string, error)
+
+// WrapCompletionFunction creates a function that can be used to register a cobra completion command that has access to the manager
+func WrapCompletionFunction(f util.Factory, ioStreams util.IOStreams, completer CobraCompletionFuncWithError) CobraCompletionFunc {
```

Comment:
I moved the completion wrapper here and exported it at it maybe can be reused by other completion functions that require the manager. Currently it is only used in the base options though

Reply:
Why not a new package genericcompletions or at least a new file completions.

Reply:
As I have already said. I think it would make sense to have first an agreed concept for this kind of PRs before doing all the implementation work.